Add IntelliJ Quarkus Tools 1.11.0 blog post (#1440)
quarkusio/quarkusio.github.io: _posts/2022-05-23-intellij-quarkus-tools-1.11.0.adoc

---
layout: post
title: 'Quarkus Tools for IntelliJ 1.11.0 released!'
date: 2022-05-23
tags: announcement ide
synopsis: 'This release adds support for Qute templates'
author: jeffmaury
---
:imagesdir: /assets/images/posts/intellij-quarkus-tools-1.11.0

We are very pleased to announce the 1.11.0 release of Quarkus Tools for IntelliJ.
This release adds support for Qute templates but also align with LSP4MP 0.4.0 and quarkus-ls 0.11.1.

== Qute templates

When a Quarkus project has Qute in the classpath, editing a Qute template (HTML / JSON / YAML / Text) will add new features into the editor.

Let's do a little tour.

=== Integration between Qute templates and Java

Now any `Template` attribute will show the option to generate a corresponding Qute template, or navigate to the existing template file.

image::quarkus-tools1.gif[]

> All Qute files paired with a corresponding Java resource supports a number of language client features, such as go to definition, hover, autocompletion, etc.. Each of these we will explore further in the article

Reviewer: ```suggestion
All Qute files paired with a corresponding Java resource support a number of language client features, such as go to definition, hover, autocompletion, etc.. Each of these we will explore further in the article
```

image::quarkus-tools2.gif[]

=== Method Parameter Support in Qute Templates

The Qute language server has support for built-in and user-defined Java classes, including parameter completion and validation for their methods. Take advantage of this to easily avoid Java related mistakes.

image::quarkus-tools3.gif[]

=== Java Type Hover Support for Qute Templates

Alongside support for Java completion and validation, the extension also supports type hovering for built-in and user-defined Java classes.

image::quarkus-tools4.gif[]

=== Qute Section Completion

> Qute templates use sections in its templating language to power any logic required when compiling and displaying the resulting document. To support this, the extension’s Qute support populates a completion list with all available Qute sections, along with any parameters they take and a small code snippet of its usage.

Reviewer: ```suggestion
Qute templates use sections in its templating language to power any logic required when compiling and displaying the resulting document. To support this, the extension’s Qute support populates a completion list with all available Qute sections, along with any parameters they take and a small code snippet of their usage.
```

image::quarkus-tools5.gif[]

=== Support for Qute Validation

Java and Qute validation is included for Qute templates with Quarkus Tools for Visual Studio Code, including diagnostics.

image::quarkus-tools6.gif[]

=== Completion for Qute Value Resolvers

> Paired with corresponding Java methods, built-in Java classes also have built-in Qute value resolvers made available by the Qute templating engine that is evaluated and outputs, or resolves to, a value. Quarkus Tools for Visual Studio Code supports completion, hover and snippet generation for these built-in value resolvers.

Reviewer: ```suggestion
Paired with corresponding Java methods, built-in Java classes also have built-in Qute value resolvers made available by the Qute templating engine that are evaluated and output, or resolve to, a value. Quarkus Tools for Visual Studio Code supports completion, hover and snippet generation for these built-in value resolvers.
```

image::quarkus-tools7.gif[]

=== Support for @TemplateExtension Methods

Template extension methods are used to extend the set of accessible properties of data objects. Qute support in Quarkus Tools for Visual Studio Code has support for marked `@TemplateExtension` Java methods, which extend a given data model.

image::quarkus-tools8.gif[]

=== Support for Directly Injecting Beans into Qute Templates

Support for Directly Injecting Beans into Qute Templates

> A CDI bean annotated with `@Named` can be referenced in any template through cdi and/or inject namespaces. Quarkus Tools for Visual Studio Code provides completion and definition navigation support for Beans.

Reviewer: The news is about Intellij and you're talking about Visual Studio Code. Is it expected?

image::quarkus-tools10.gif[]

=== MicroProfile Fault Tolerance

We’ve added basic validation support for `@Bulkhead`, `@Timeout`, `@Asynchronous`, and `@Scheduled` annotations.

image::quarkus-tools11.gif[]

=== Application Path support

When launching a Quarkus application, the code lens URL endpoints now supports `@ApplicationPath` and will take its value into account.

image::quarkus-tools12.png[]

In this example, the `@ApplicationPath` is set to `/api`, and the code on the method declaration (below) takes that into account when generating the URL.

image::quarkus-tools13.png[]

=== Completions

At the end of the day, everyone wants to spend less time trying to hunt down some obscure setting, and more time developing the logic.

If you’re using the `@CacheResult` annotation on a method in your Java sources, we automatically suggest the settings available to configure it from your configuration file. It’s also fairly reactive to changes so you’ll always know if something has gone wrong.

image::quarkus-tools14.gif[]

> `@ConfigMapping` is now also supported with some basic validation as well as completions for the application properties.

Reviewer: ```suggestion
`@ConfigMapping` is now also supported with some basic validation as well as completion for the application properties.
```
